pkg/envoy/secretsync.go

```diff
@@ -90,8 +90,13 @@ func (r *secretSyncer) upsertK8sSecretV1(ctx context.Context, secret *slim_corev
 		return errors.New("secret is nil")
 	}
 
+	envoySecret := k8sToEnvoySecret(secret)
+	if envoySecret == nil {
+		return nil
+	}
+
 	resource := Resources{
-		Secrets: []*envoy_extensions_tls_v3.Secret{k8sToEnvoySecret(secret)},
+		Secrets: []*envoy_extensions_tls_v3.Secret{},
```

Comment:
Do you mean this? Or am I missing something?

```suggestion
		Secrets: []*envoy_extensions_tls_v3.Secret{envoySecret},
```

Reply:
Yep of course :smile: :facepalm: Thanks!